modules/apps/frontend-js/frontend-js-svg4everybody-web/src/main/java/com/liferay/frontend/js/svg4everybody/web/internal/servlet/taglib/SVG4EverybodyTopHeadDynamicInclude.java

```diff
@@ -64,7 +64,22 @@ public void include(
 			}
 		}
 
-		if (!cdnDynamicResourcesEnabled) {
+		boolean cdnHostEnabled = false;
+
+		try {
+			String cdnHost = _portal.getCDNHost(httpServletRequest);
+
+			cdnHostEnabled = !cdnHost.isEmpty();
+		}
+		catch (PortalException portalException) {
+			if (_log.isWarnEnabled()) {
+				_log.warn(
+					"Unable to gather property from configuration",
+					portalException);
+			}
+		}
+
+		if (cdnHostEnabled) {
```

Comment:
What is the relationship between CDN host being enabled and writing the <script> nodes? :thinking: 

Why don't we write them when there's no CDN?

Reply:
When there's cdn, the hosts are different, so `foo.com` will attempt to load the sprite from `cdn.foo.com`. That results in a CORS error because `svg use` does not support it. 

For that reason, when CDN is enabled and we expect icons can be loaded from there we enable `svg4everybody` which will retrieve them via an AJAX request which does support it.

Reply:
:-o Magic...

Reply:
In that case I think this is the way to do it. There's no point in internalizing such logic inside `AbsolutePortalURLBuilder` :+1: